### PR TITLE
WEA-10 Check if location services is turned on

### PR DESCRIPTION
**Pull request by ayush-shta**

### CODE REVIEW

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 1]
```diff
@@ -60,6 +61,9 @@
   }
 
   private void fetchCurrentLocation() {
+
+    LocationUtils.checkLocationServices(view.getContext());
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: this is a void method and name looks/needs a boolean. If location is not provided a dialog is shown but still location request object is created.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -12,4 +12,6 @@
   <string name="details_caps">DETAILS</string>
 
   <string name="request_permission_location">This app needs access to your location  to know where you are</string>
+  <string name="gps_network_not_enabled">Please enable GPS network</string>
```
Comment (subash-poudel) [inline app/src/main/res/values/strings.xml]: is gps a network?

File: app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import android.support.v7.app.AlertDialog;
+import com.ayush.weatherapp.R;
+
+public final class LocationUtils {
+
+  private LocationUtils() {
+  }
+
+  //Reference: https://stackoverflow.com/questions/10311834/how-to-check-if-location-services-are-enabled
+  public static void checkLocationServices(Context context) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java]: The dialog could be separated into it's own function and created in the activity.

File: app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    gpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+    networkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+
+    if (!gpsEnabled && !networkEnabled) {
+      // notify user
+      AlertDialog.Builder dialog = new AlertDialog.Builder(context);
+      dialog.setMessage(context.getResources().getString(R.string.gps_network_not_enabled));
+      dialog.setPositiveButton(context.getResources().getString(R.string.open_location_settings),
+          (paramDialogInterface, paramInt) -> {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java]: param is retundant

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (ayush-shta) [reply app/src/main/res/values/strings.xml]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java (hunk 4 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java (hunk 1 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java (hunk 3 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java]: :ok:

**Pull request merged**